R/combinedscatterplot.R

```diff
@@ -899,6 +906,18 @@ fitLines <- function(scatter.colors, scatter.colors.as.categorical, scatter.grou
          fit.ci.label.colors = fit.ci.label.colors)
 }
 
+extractUniqueValues <- function(groups, not.na)
+{
+    if (is.factor(groups))
+    {
+        lvls <- levels(groups)
+        return(lvls[lvls %in% groups[not.na]])
+    }
+    else if (any(class(groups) %in% c("Date", "POSIXct", "POSIXt", "integer", "numeric")))
+        return(sort(unique(groups[not.na])))
+    else
+        return(unique(groups[not.na]))
+}
```

Comment:
Nice! Could you also call this function to create `g.list` inside `processAnnotations`?